Feed Parser

olaydub opened the pull request:
Feed parser to parse XML

olaydub committed:
add

404 error and MIME type exception for parser.js occuring

olaydub committed:
add parser.js to webpack.config.js

attempt to solve 404 and MIME type error in web console

olaydub committed:
fix webpack

changed js files to ts, webpack fix to actually copy correct files to dist and resolve errors with Getting parser.js

wandyezj commented on package.json:
```diff
@@ -29,15 +29,17 @@
     },
     "homepage": "https://github.com/wandyezj/website#readme",
-    "dependencies": {},
+    "dependencies": {
```
Why is this dependency needed?

wandyezj commented on package.json:
```diff
@@ -29,15 +29,17 @@
         "clean-webpack-plugin": "^4.0.0",
         "copy-webpack-plugin": "^7.0.0",
         "html-webpack-plugin": "^5.5.0",
+        "playwright": "^1.33.0",
```
Why did these dev dependencies move?

wandyezj commented on src/index.js:
```diff
@@ -0,0 +1,16 @@
+const express = require('express');
```
You shouldn't need this index.js file.

wandyezj commented on src/parser.ts:
```diff
@@ -0,0 +1,17 @@
+//retrieve RSS feed from specified URL, processes response, logs parsed XML
+//data to console
+export function fetchAndParseRSS(RSS_URL : string) {
+    fetch(RSS_URL)
```
It's a little easier to use `async` `await` instead of then chains. see [async and await on MDN](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises#async_and_await)

olaydub committed:
run style

npm run style to hopefully pass checks

olaydub committed:
parse specific fields

- The title of the article
- The date/time it was published
- The image (or main image if multiple) associated with it
- The content (description)
- Link to the article

wandyezj commented on webpack.config.js:
```diff
@@ -54,10 +54,26 @@ module.exports = async (env, options) => {
                     },
+                    {
+                        to: "index.ts",
+                        from: "./src/index.ts",
```
Why is this copied?

wandyezj commented on src/parser.ts:
```diff
@@ -0,0 +1,36 @@
+
+//retrieve RSS feed from specified URL, processes response, logs parsed XML
+//data to console
+export function fetchAndParseRSS(RSS_URL: string) {
```
`fetchAndParseRSS` should have a unit test

Jquin997 committed:
fix bugs & add unit test for parser.ts

Jquin997 committed:
Update build.yml

set server starts before test begin

Jquin997 committed:
Update parser.test.ts

Jquin997 committed:
solve page.evaluate issue

Jquin997 committed:
update  robust and maintainable tests

Jquin997 committed:
Refine tests for RSS feed parsing

Jquin997 committed:
update feed img test

wandyezj committed:
patch

wandyezj committed:
merge

wandyezj committed:
merge

wandyezj commented on .idea/workspace.xml:
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Can we do this in a separate PR?

wandyezj committed:
cleanup

wandyezj committed:
update build

wandyezj committed:
test example
